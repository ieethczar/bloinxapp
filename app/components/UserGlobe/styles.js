import { StyleSheet } from "react-native";

import * as theme from '../../theme';

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: 5,
  },
  text: {
    fontSize: 10,
    color: theme.grayText,
  }
});

export default styles;