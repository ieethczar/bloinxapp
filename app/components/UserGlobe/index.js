import React from 'react';
import PropTypes from 'prop-types';
import { View, TouchableOpacity } from 'react-native';
import { Avatar, AirbnbRating, Text } from 'react-native-elements';

import styles from './styles';

const UserGlobe = ({ size, name, image, onPress, rating, ratingSize = 9 }) => {
  const Content = onPress ? TouchableOpacity : View;
  return (
    <Content style={styles.container}>
      <Avatar
        size={size}
        rounded
        source={{ uri: image }}
      />
      <View style={styles.rating}>
        <AirbnbRating
          size={ratingSize}
          defaultRating={rating}
          showRating={false}
          isDisabled
        />
      </View>
      <Text style={styles.text}>{name && name.substring(0,10)}</Text>
    </Content>
  );
}

UserGlobe.defaultProps = {
  size: 'small',
  name: '',
  image: 'https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg'
};

UserGlobe.propTypes = {
  size: PropTypes.string.isRequired,
  name: PropTypes.string,
  image: PropTypes.string,
  onPress: PropTypes.func,
  rating: PropTypes.number,
  ratingSize: PropTypes.number,
};

export default UserGlobe;