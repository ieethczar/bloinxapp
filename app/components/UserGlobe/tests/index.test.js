import React from 'react';
import renderer from 'react-test-renderer';

import UserGlobe from '../index';

describe('UserGlobe', () => {
  it('', () => {
    const component = renderer.create(
      <UserGlobe
        size="small"
        name="Text"
        image="some-image-url"
        onPress={jest.fn()}
        rating={5}
        ratingSize={5}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});