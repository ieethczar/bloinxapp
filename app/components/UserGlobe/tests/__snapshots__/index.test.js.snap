// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGlobe  1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "opacity": 1,
      "padding": 5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 17,
        "height": 34,
        "width": 34,
      }
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 17,
          "flex": 1,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        borderRadius={17}
        onLoad={[Function]}
        source={
          Object {
            "uri": "some-image-url",
          }
        }
        style={
          Object {
            "bottom": 0,
            "flex": 1,
            "height": undefined,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": undefined,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "flex": 1,
              "height": undefined,
              "justifyContent": "center",
              "width": undefined,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 5,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 5,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 5,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 5,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 5,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 5,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 5,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 5,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 5,
                "margin": 3,
                "tintColor": "#f1c40f",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 5,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#555555",
        "fontSize": 10,
      }
    }
  >
    Text
  </Text>
</View>
`;
